### ISSUE TITLE
Map timezones for networks properly

### ISSUE THREAD

**Issue by ialokim** [grote/Transportr#577]
**Describe the bug**
The time for trip queries is not properly mapped from the device's timezone to the network's timezone.

**To Reproduce**
1. Set phone timezone to Germany
2. Select Nicaragua as Network
3. Search for a connection at 9 PM (on the phone = in Germany)
4. Note that the next connections are shown to be at 6 AM

**Expected behavior**
As in Nicaragua, it is merely 1 or 2 PM (depending on daylight saving time in Germany), there are indeed a lot of public transport services still serving until evening, which should be displayed.

I didn't check if the same happens with other networks and timezones too. Could be related to #382, but should be a lot easier to fix as Nicaragua covers one timezone only.

**Transportr Version:** 2.0.5

**Comment by grote** [comment]
Maybe related to https://groups.google.com/forum/#!topic/public-transport-enabler-discuss/-rco45pcfcg ?

**Comment by ialokim** [comment]
It certainly is. FYI, [Navitia docs](https://doc.navitia.io/#iso-date-time) state the following:
> Navitia
>
>exposes every date times as local times of the coverage via an ISO 8601 “YYYYMMDDThhmmss” string
>
>can be requested using local times of the coverage via ISO 8601 as “YYYYMMDDThhmmss” or “YYYY-MM-DDThh:mm:ss”
>
>can be requested using UTC relative times via ISO 8601 as “YYYYMMDDThhmmss+HHMM” or “YYYY-MM-DDThh:mm:ss+HH:MM”
>
>can be requested using UTC times via ISO 8601 as “YYYYMMDDThhmmssZ” or “YYYY-MM-DDThh:mm:ssZ”
>
>Context object provides the Timezone, useful to interpret datetimes of the response.

So there are probably two seperate issues here:
* `AbstractNavitiaProvider` should use the UTC relative time format for requests
* Transportr should convert the returned local time of the coverage to the phone's one.

**Comment by ialokim** [comment]
> Maybe related to https://groups.google.com/forum/#!topic/public-transport-enabler-discuss/-rco45pcfcg ?

Looking at the answer in this discussion, it seems PTE is expected to return all times in the UTC timezone (?):
> The timeZone property is meant so that the parser can convert any time
to UTC (if the API doesn't already return UTC -- newer APIs do if I
remember right). 

Not sure whether I understood it correctly, but in this case `AbstractNavitiaProvider` should convert all times to UTC (which it is not doing at the moment) and Transportr should adopt it's logic to convert it to the phone's timezone.

Maybe you could clarify this for me @grote and @schildbach?

**Comment by schildbach** [comment]
Yes, the API is UTC time for both request and response. In your app, you should convert the time values into either local timezone or the timezone returned by provider.getTimeZone().

**Comment by ialokim** [comment]
Ok, thanks for the clarification, so there are some changes to be done :smile:

I've opened an issue in schildbach/public-transport-enabler#260 mentioning the necessary changes for `AbstractNavitiaProvider`.

**Comment by grote** [comment]
Strange that this works (in most cases) in Transportr without the conversation.

**Comment by ialokim** [comment]
This should be fixed with schildbach/public-transport-enabler#261.

> Strange that this works (in most cases) in Transportr without the conversation.

It works without conversion as the `Date` object returned always use UTC times and it was implemented as expected e.g. for Hafas providers. In the case of Navitia based providers, it worked when using it while being in the same timezone the provider uses (as there was no conversion done and it wasn't necessary too).

I would keep this issue open until we confirmed it's solution with a new PTE version.
